src/components/TodoHeader.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState, useEffect } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+
+interface TodoHeaderProps {
+  todos: Todo[];
+  onAddTodo: (title: string) => Promise<void>;
+  onToggleAll: () => Promise<void>;
```

Comment:
it is better to type these functions as synchronus

```suggestion
  onAddTodo: (title: string) => void;
  onToggleAll: () => void;
```